rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -291,10 +291,14 @@ class LabelBatch : public MetaDataBatch
     }
     std::shared_ptr<MetaDataBatch> clone(bool copy_contents) override
     {
-        if(copy_contents)
+        if(copy_contents) {
             return std::make_shared<LabelBatch>(*this);
-        else
-            return std::make_shared<LabelBatch>();
+        } else {
+            std::shared_ptr<MetaDataBatch> label_batch_instance = std::make_shared<LabelBatch>();
+            label_batch_instance->resize(this->size());
+            label_batch_instance->get_info_batch() = this->get_info_batch();
```

Comment:
So when copy data is set to false
we will just resize the labels_batch to batchsize and fill the info for labels required
then somewhere else we will fill the contents, is that what happening here?

Reply:
Yes